hip/matrix/dense_kernels.hip.cpp

```diff
@@ -77,7 +77,80 @@ constexpr int default_block_size = 512;
 
 
 template <typename ValueType>
-void simple_apply(std::shared_ptr<const HipExecutor> exec,
+void compute_dot_dispatch(std::shared_ptr<const DefaultExecutor> exec,
+                          const matrix::Dense<ValueType>* x,
+                          const matrix::Dense<ValueType>* y,
+                          matrix::Dense<ValueType>* result)
+{
+    if (x->get_size()[1] == 1 && y->get_size()[1] == 1) {
+        if (hipblas::is_supported<ValueType>::value) {
+            auto handle = exec->get_hipblas_handle();
+            hipblas::pointer_mode_guard pm_guard(handle);
+            hipblas::dot(handle, x->get_size()[0], x->get_const_values(),
+                         x->get_stride(), y->get_const_values(),
+                         y->get_stride(), result->get_values());
+        } else {
+            GKO_NOT_IMPLEMENTED;
```

Comment:
I don't think that's a good alternative, how about deferring to compute_dot here as well?